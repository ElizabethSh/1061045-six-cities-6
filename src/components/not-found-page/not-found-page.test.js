import React from 'react';
import {render} from "@testing-library/react";
import {createMemoryHistory} from "history";
import {Router} from "react-router";
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import NotFoundPage from './not-found-page';

const mockStore = configureStore({});
let history;

describe(`Test NotFoundPage`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });

  it(`NotFoundPage should render correctly, when user is logged in`, () => {
    const store = mockStore({
      USER: {isLoggedIn: true, usersEmail: `test@gmail.com`}
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <NotFoundPage />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`NotFoundPage should render correctly, when user is not logged in`, () => {
    const store = mockStore({
      USER: {isLoggedIn: false, usersEmail: null}
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <NotFoundPage />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});

