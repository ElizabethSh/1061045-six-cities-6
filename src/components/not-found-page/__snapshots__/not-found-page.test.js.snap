// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NotFoundPage NotFoundPage should render correctly, when user is logged in 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
                style="display: flex;"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    test@gmail.com
                  </span>
                </a>
                <button
                  style="margin-left: 10px;"
                >
                  Log Out
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main>
      <section
        style="margin: 150px auto; width: 420px; text-align: center;"
      >
        <h1
          style="padding: 20px 0px 80px;"
        >
          404: Page is not found
        </h1>
        <a
          class="button form__submit"
          href="/"
        >
          Go back to the main page
        </a>
      </section>
    </main>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;

exports[`Test NotFoundPage NotFoundPage should render correctly, when user is not logged in 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
                style="display: flex;"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main>
      <section
        style="margin: 150px auto; width: 420px; text-align: center;"
      >
        <h1
          style="padding: 20px 0px 80px;"
        >
          404: Page is not found
        </h1>
        <a
          class="button form__submit"
          href="/"
        >
          Go back to the main page
        </a>
      </section>
    </main>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;
